testing/hooks_test.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+import { afterEach, beforeEach, withHooks } from "./hooks.ts";
+import { assert, assertEquals, AssertionError } from "./asserts.ts";
+
+Deno.test("beforeEach is exectued correctly", function (): void {
+  let didInvoke = false;
+  beforeEach(function () {
+    didInvoke = true;
+    assertEquals(/deno/, /deno/);
+  });
+  assertEquals(didInvoke, false);
+  withHooks(function () {
+    assertEquals(/cat/, /cat/);
```

Comment:
I think we should assert `assertEquals(didInvoke, true)` here to check `beforeEach` hook is actually called before the main function.

Reply:
Yes , I will `assertEquals(didInvoke, true)` inside withHooks